### DISCUSSION TITLE
⏳ 🔗 ⛔ Revamp of the mobile menu / add MANY more styling options to the new "menu" widget for mobile!

### DISCUSSION THREAD

**Discussion by SolunaCG** [elementor/elementor#23634] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Hey Elementor community!

Since #3275 was closed 11 hours ago I decided to add a new request to make sure the things I listed there don´t get lost. 

As I´ve already mentioned a couple of times here on github I think the Elementor mobile menu is BY FAR the weakest element when working with Elementor. Nowadays where "mobile first" is a very common design concept I´d say that the mobile menu is a really important element of the whole design. 

The current mobile menu that we have with Elementor is so minimalistic with so few design possibilities that imo you always have to rely on other plugins like the mobile menu from Crocoblock (https://crocoblock.com/widgets/hamburger-menu/), ThePlus (https://theplusaddons.com/widgets/mobile-menu/), Unlimited Elements (https://unlimited-elements.com/fullscreen-menu-widget-for-elementor/) or other plugin developers which all work better or worse. 

Like I´ve just mentioned: I have no idea why this hasn´t been requested more by other users and in addition to this request I´d love to know how other users create their mobile menus. Our clients have NEVER been satisfied so far with how the standard Elementor mobile menu looks like. They´ve always wanted more advanced designs and some other things I´m going to mention below which are part of this request.

### Describe the solution you'd like

What I´d love to see when it comes to the Elementor mobile menu:

- option to load templates within the menu (-> Crocoblock hamburger menu)
- multiple animation presets for the menu container (slide in (ideally from top, left, bottom or right), zoom in, fade in, etc.)
- multiple animation presets for the menu toggle icon (-> https://unlimited-elements.com/animated-hamburger-icon-for-elementor/)
- easy way to make the menu full width and full height for all responsive devices
- NO responsive settings like we already have them in the "Tabs"-widget or the new megamenu! Please get rid of the automatic responsive design for mobile devices! That prevents us users from freely designing the menu for those devices.

Just as a few examples of better mobile menus: (Check these websites on your phone)

https://powerpackelements.com/ :
Animated Hamburger Icon, nice slide-down effect for the menu, slide-down effect for the sub-menu items.

https://bricksbuilder.io/ :
Animated Hamburger Icon, full Height and full Width, very cool sub-menu with (probably) nested Elements. And Bricks even does all that by default without plugins. So I´d love to see this in Elementor, too!

https://unlimited-elements.com/liquid-fullscreen-menu-widget-for-elementor/ :
Animated Hamburger Icon, fancy styling with the "liquid" effect (probably a bit too much but just as ideas what you COULD add to make it look better) and what I like is that the menu items appear one after another from bottom to top. 

In fact, I never use the Elementor Mega Menu or the old Elementor Menu because my clients have always complained about how bad it looks. So I´ve always added 3rd party plugins just for a better mobile menu. And imo that shouldn´t be necessary. 
It´s hard to tell the clients why we can easily handle complex dynamic data or create complex websites with loop grids or whatever but we can NOT create a good mobile menu for them. 

For my clients and me this is not just a feature that would be nice to have. but something that is ESSENTIAL!

### Describe alternatives you've considered

Unfortunately the only alternatives are external plugins like the ones that I´ve mentioned above but even those are very limited in terms of designing. 

### Additional context

Imo the most important things the mobile menu needs are: 

- make it nested and let us freely design the mobile menu!
- add multiple animation presets for the menu toggle icon!

**Comment by sanholo177** [comment]
I second this! Elementor needs to give its menu a proper lift. At the moment you can't go full-screen menu on mobile without using a Pop-up. I would love to see nested capabilities in the Mobile Menu. As well was the Slide option for Sub-menus instead of vertical drop-downs. This can work well for Menus with multiple children. But, at the end of the day, all that matters is if/when Elementor listens to us and focuses on working on things like this. 

Instead, the team focuses on rolling out AI stuff.

**Comment by spasauna** [comment]
Please make this happen.

**Comment by codemweb** [comment]
Very this!
